platform/default/bidi.cpp

```diff
@@ -1,9 +1,25 @@
-#include <memory>
-
 #include <mbgl/text/bidi.hpp>
+
 #include <unicode/ubidi.h>
 #include <unicode/ushape.h>
 
+#include <memory>
+#include <type_traits>
+
+namespace {
+
+template <typename To, typename From>
+typename std::enable_if_t<std::is_convertible<From, To>::value, To> convertCharPtr(From data) {
+    return data;
+}
+
+template <typename To, typename From>
+typename std::enable_if_t<!std::is_convertible<From, To>::value && sizeof(To) == sizeof(From), To> convertCharPtr(From data) {
```

Comment:
Some code you can borrow for this: https://github.com/jfirebaugh/PlusPlus/blob/master/Code/PlusPlus/Standardish/charlike.h